documentation/specs/proposed/BuildCheck-Architecture.md

```diff
@@ -47,6 +47,25 @@ The BuildCheck infrastructure will be prepared to be available concurrently with
 * BuildCheck will need to recognize custom analyzers packages during the evaluation time - so some basic code related to BuildCheck will need to be present in the worker node.
 * Presence in worker node (as part of the `RequestBuilder`), will allow inbox analyzers to agile leverage data not available within `BuildEventArgs` (while data proven to be useful should over time be exposed to `BuildEventArgs`)
 
+## Replay Mode
+
+Prerequisites: [MSBuild Replaying a Binary Log](../../wiki/Binary-Log.md#replaying-a-binary-log)
+
+When replaying a binary log, we can apply BuildCheck with help of `-analyze` switch:
+```
+> msbuild.exe msbuild.binlog -analyze
+```
+
+If BuildCheck is enabled, then the events from `BinaryLogReplayEventSource` and new events from BuildCheck are merged into the `IEventSource`, from which the loggers get events.
+
+<img src="https://github.com/surayya-MS/msbuild/assets/114938397/7f24f5ee-f254-41a1-bf92-0c476ca8b90e" width="40%" height="40%">
```

Comment:
IS it something that needs to be changed after it is merged? Is there a way to make this image a part of the documentation and reference it from the documentation?

Reply:
We shouldn't check in images - instead, we should checking [Mermaid diagrams](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams), which are a text-based representation of charts/graphs that GitHub will render for us.

Reply:
I'd see an image as acceptable - but it needs to live collocated with the document (so it'd need to be part of this PR).

If possible - mermaid diagrams are definitely preferable (but still image is better than no diagram at all :-))


We have both cases in our repo

Reply:
Thanks! I used mermaid diagram